hedera-node/hedera-app/src/main/java/com/hedera/node/app/ServicesMain.java

```diff
@@ -228,12 +228,14 @@ public static void main(final String... args) throws Exception {
                     "Multiple nodes were supplied via the command line. Only one node can be started per java process.");
             exitSystem(NODE_ADDRESS_MISMATCH);
         }
-        final List<NodeId> nodesToRun = getNodesToRun(diskAddressBook, commandLineArgs.localNodesToStart());
-        checkNodesToRun(nodesToRun);
-        final var selfId = ensureSingleNode(nodesToRun, commandLineArgs.localNodesToStart());
         final var platformConfig = buildPlatformConfig();
+        final var cliNodesToRun = commandLineArgs.localNodesToStart();
+        final var configNodesToRun =
+                platformConfig.getConfigData(BasicConfig.class).nodesToRun();
+        final List<NodeId> nodesToRun = getNodesToRun(diskAddressBook, cliNodesToRun, configNodesToRun);
+        final var selfId = ensureSingleNode(nodesToRun);
```

Comment:
😵‍💫 this took some digging. For clarity, especially when I read this code again next, I suggest:

```suggestion
        // Determine which nodes were _requested_ to run from the command line
        final var cliNodesToRun = commandLineArgs.localNodesToStart();
        // Determine which nodes are _configured_ to run from the config file(s)
        final var configNodesToRun =
                platformConfig.getConfigData(BasicConfig.class).nodesToRun();
        // Using the requested nodes to run from the command line, the nodes configured to run, and now the
        // address book on disk, reconcile the final list of nodes to run
        final List<NodeId> nodesToRun = getNodesToRun(diskAddressBook, cliNodesToRun, configNodesToRun);
        // Finally, verify that the reconciliation of above node IDs yields exactly one node to run
        final var selfId = ensureSingleNode(nodesToRun);
```